drivers/d3d12/d3d12_hooks.h

```diff
@@ -34,11 +34,21 @@
 
 class D3D12Hooks {
 private:
-	static D3D12Hooks *singleton;
+	static inline D3D12Hooks *singleton = nullptr;
 
 public:
-	D3D12Hooks();
-	virtual ~D3D12Hooks();
+	D3D12Hooks() {
+		if (singleton == nullptr) {
+			singleton = this;
+		}
+	}
+
+	virtual ~D3D12Hooks() {
+		if (singleton == this) {
+			singleton = nullptr;
+		}
+	}
```

Comment:
This might be questionable, but these are very simple and were the only thing in the cpp file, so it could be removed...

Reply:
I don't think removing `.cpp` files is a _goal_, but having this in the `.h` file certainly isn't wrong either. I think it's fine :)